### PR TITLE
expand fsspec downstream connectors

### PR DESCRIPTION
**Pull request by rbiseck3**
### Description
Replacing PR [1383](https://github.com/Unstructured-IO/unstructured/pull/1383)

### CODE REVIEW

File: unstructured/ingest/interfaces.py [hunk 1]
```diff
@@ -132,6 +132,13 @@ def __post_init__(self):
             self.file_path = ""
             return
 
+        # dropbox paths can start with slash
```
Comment (ryannikolaidis) [inline unstructured/ingest/interfaces.py:135]: oh fun. good catch

File: test_unstructured_ingest/test-ingest-box-dest.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+#    --verbose \
+#    --reprocess \
+#    --input-path example-docs/fake-memo.pdf \
+#    --work-dir "$WORK_DIR" \
+#    box \
+#    --box-app-config "$BOX_APP_CONFIG_PATH" \
+#    --remote-url "$DESTINATION_BOX" \
+#
+## Simply check the number of files uploaded
+#expected_num_files=1
```
Comment (ryannikolaidis) [inline test_unstructured_ingest/test-ingest-box-dest.sh:54]: do we not support box yet? leaving this in for when we do? ...if so, would leave details at the top

File: test_unstructured_ingest/test-ingest-box-dest.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+#TODO currently box api/sdk does not work to create folders and check for content similar to other fsspec ingest tests
```
Comment (ryannikolaidis) [inline test_unstructured_ingest/test-ingest-box-dest.sh:2]: for the purposes of testing: can we just attempt to download the content in the test and confirm that we can fetch it?

Comment (rbiseck3) [reply test_unstructured_ingest/test-ingest-box-dest.sh:2]: The problem here is I can't even create a temporary path with confidence to use for testing in the same way I create it everywhere else. We could technically push to a new path but then have no way to clean it up at the end, bloating how much we store in the box account.

**Review by ryannikolaidis — COMMENTED** [review]
looks good. a couple quick questions above otherwise gtg

File: test_unstructured_ingest/test-ingest-gcs-dest.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    --reprocess \
+    --input-path example-docs/fake-memo.pdf \
+    --work-dir "$WORK_DIR" \
+    gcs \
+    --token "$GCP_INGEST_SERVICE_KEY_FILE" \
+    --remote-url "$DESTINATION_GCS"
+
+# Simply check the number of files uploaded
+expected_num_files=1
+num_files_in_gcs=$(gcloud storage ls "$DESTINATION_GCS"/example-docs/ | wc -l )
```
[the remark this answers is not in the document]
Comment (rbiseck3) [reply test_unstructured_ingest/test-ingest-gcs-dest.sh:57]: Created one here: [drop dependency on cloud CLIs in CI](https://github.com/Unstructured-IO/unstructured/issues/1941)

[the remark this answers is not in the document]
Comment (rbiseck3) [reply test_unstructured_ingest/test-ingest-gcs-dest.sh:57]: Will also have to investigate how to clean up the created folders without the CLIs.

File: test_unstructured_ingest/test-ingest-box-dest.sh (hunk 3 above)
Comment (rbiseck3) [reply test_unstructured_ingest/test-ingest-box-dest.sh:2]: Well across the board, it's aws cli for s3, `az` for azure and `gcloud` for google gloud storage.

Comment (rbiseck3) [reply test_unstructured_ingest/test-ingest-box-dest.sh:2]: For dropbox it had to be done through their api using `curl`

File: test_unstructured_ingest/test-ingest-gcs-dest.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (ryannikolaidis) [reply test_unstructured_ingest/test-ingest-gcs-dest.sh:57]: Perfect, thanks!

**Review by ryannikolaidis — APPROVED** [review]
Lgtm!

**Pull request merged**